cmd/minikube/cmd/start_flags.go

```diff
@@ -276,6 +276,15 @@ func generateClusterConfig(cmd *cobra.Command, existing *config.ClusterConfig, k
 			glog.Errorf("Found deprecated --enable-default-cni flag, setting --cni=bridge")
 			chosenCNI = "bridge"
 		}
+		// networkPlugin cni deprecation warning
+		chosenNetworkPlugin := viper.GetString(networkPlugin)
+		if chosenNetworkPlugin == "cni" {
+			out.WarningT("Found deprecated --network-plugin=cni flag (new flag: --cni), setting --cni=auto if not already set")
```

Comment:
I misspoke in the original bug report, the actual deprecated flag was --enable-default-cni, checked above this. --network-plugin is still a necessary evil for some cases.

Could you change the warning to say something more like:

`With --network-plugin=cni, you will need to provide your own CNI. See --cni flag as a user-friendly alternative`

I would minikube not suddenly change the behavior of existing users of `--network-plugin=cni` if it can be avoided.

Reply:
I've just updated the code.